features/polyExplorer/src/components/embeddedSankey/embeddedSankey.jsx

```diff
@@ -0,0 +1,123 @@
+import React, { useState } from "react";
+import { Chips } from "@polypoly-eu/poly-look";
+import { PolyChart } from "@polypoly-eu/poly-look";
+
+const jurisdictionColors = {
+    "EU-GDPR": "#60E6DE",
+    "Five-Eyes": "#EC453D",
+    Russia: "#FE8988",
+    China: "#C5271E",
+    Sonstige: "#A9B6C6",
+};
+
+const transparent = "rbga(0, 0, 0, 0)";
+
+const inactiveLabelOpacity = 0.3;
+
+const gradients = Object.entries(jurisdictionColors).map(
+    ([jurisdiction, color]) => ({
+        id: jurisdiction + "-gradient",
+        stops: [
+            { offset: "0%", color, opacity: 0.2 },
+            { offset: "100%", color, opacity: 1 },
+        ],
+    })
+);
+
+function linksToGroups(links) {
+    return {
+        sources: [...new Set(links.map((e) => e.source))],
+        targets: [...new Set(links.map((e) => e.target))],
+    };
+}
+
+const EmbeddedSankey = ({ links, groups }) => {
+    const { sources, targets } = linksToGroups(links);
+    const [activeSources, setActiveSources] = useState(
+        groups?.source?.all ? sources : [sources[0]]
+    );
+    const [activeTargets, setActiveTargets] = useState(
+        groups?.target?.all ? targets : [targets[0]]
+    );
+
+    const handleActiveSourceChange = (source) => {
+        if (source === "all") setActiveSources(sources);
+        else if (source === "others") setActiveSources(groups.source.others);
+        else setActiveSources([source]);
+    };
+
+    const handleActiveTargetChange = (target) => {
+        if (target === "all") setActiveTargets(targets);
+        else if (target === "others")
+            return setActiveTargets(groups.target.others);
+        setActiveTargets([target]);
+    };
```

Comment:
This code is very similar to the code above, except for variable and function names. Wonder if it could be refactored somehow... Essentially we're repeating the same thing for sources and targets over and over.

Reply:
true

Reply:
`    const handleActiveEntityChange = (entity, allEntities, setState, others) => {
        if (entity === "all") setState(allEntities);
        else if (entity === "others") setState(others);
        else setState([entity]);
    };`
Would you prefer this? I feel that the readability decreased by a lot

Reply:
I also cannot find a better name for source/target than entity

Reply:
No, not really. Let's leave it like it is for the time being.